src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/DataReaderTest/DataReaderTest.cs

```diff
@@ -605,6 +605,56 @@ integrated into a comprehensive development
             }
         }
 
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public static async Task CanReadSequentialDecreasingChunks()
+        {
+            const string baseString = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+            StringBuilder inputBuilder = new StringBuilder();
+            while (inputBuilder.Length < (64 * 1024))
+            {
+                inputBuilder.Append(baseString);
+                inputBuilder.Append(' ');
+            }
+
+            string input = inputBuilder.ToString();
+
+            StringBuilder resultBuilder = new StringBuilder();
+            CancellationTokenSource cts = new CancellationTokenSource();
+            using (var connection = new SqlConnection(DataTestUtility.TCPConnectionString))
```

Comment:
More curiosity than request: for unit tests, I'm usually more liberal with usage of using statements vs blocks, so I'm wondering if your preference for the blocks is to trigger a disposal before running assertions or if behavior has no impact here and it's just a style preference.

Reply:
Purely style. It would work with blocks the same way as with statements. I learnt c# at the first version and there are some newer idioms that I simply prefer not to use. In general I like to have very clear scoped blocks to help me visually understand what's in and out of scope.

Reply:
Understood - I agree, there's some newer features I don't plan on using (eg, auto-constructor assignment of member variables). Just wanted to make sure the blocks weren't integral to the behavior of the test.